### PR TITLE
switch to new agent image

### PR DESCRIPTION
**Pull request by compojoom**
The new beamer executable is no longer called `beamer-agent`, but just `beamer`. It currently has 2 commands `agent` and `health-check`. The `agent` command behaves as the previous `beamer-agent` command.

The `health-check` command is used to analyze the beamer contracts and report problems such as - missed fills, missed claims or report on ongoing challenges. It also can report the agent’s balance.

### CODE REVIEW

**Comment by compojoom** [comment]
@bilbeyt can you try to follow the instructions now, please?

File: data/beamer-health-check.conf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[default]
```
Comment (bilbeyt) [inline data/beamer-health-check.conf:1]: [default] should be removed

File: data/beamer-health-check.conf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+agent-address=""
+deployment-dir="../deployments/mainnet"
```
Comment (bilbeyt) [inline data/beamer-health-check.conf:3]: should change it to directly /data/deployments/mainnet

File: data/beamer-health-check.conf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+request-throttling-in-sec=60
+
+[notification.telegram]
+token=""
+chat-id=""
+request-throttling-in-sec=0
+
+[chains.arbitrum]
+rpc-url=""
+explorer=https://arbiscan.io/tx/
```
Comment (bilbeyt) [inline data/beamer-health-check.conf:19]: these should be wrapped with "" also true for optimism explorer

**Comment by bilbeyt** [comment]
i think the same config template is in beamer main also so you should fix that config too

**Comment by manuelwedler** [comment]
We just discussed to have the documentation about the notification system only in the docs. See https://github.com/beamer-bridge/beamer/pull/1590#discussion_r1144844001

I already added it to my PR. Could you remove [this part](https://github.com/beamer-bridge/run-your-own-agent/pull/17/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R159-R182) from the README and instead add a link to https://docs.beamerbridge.com/configuration.html#notification-system ?

File: README.md [hunk 4]
```diff
@@ -60,21 +63,20 @@ If it is not, you can apply for whitelisting at hello@beamerbridge.com.
...
 1. Edit `data/agent-mainnet.conf` and make sure that the following keys have correct values:
 
-   - `[account.path]` - the path to your JSON keystore file
-   - `[account.password]` - the password to unlock your keystore file
-   - `[chains.l1.rpc-url]` - the RPC endpoint to use for mainnet Ethereum (e.g. `https://mainnet.infura.io/v3/ID`)
-   - `[chains.arbitrum.rpc-url]` - the RPC endpoint to use for mainnet Arbitrum (e.g. `https://arb1.arbitrum.io/rpc`)
-   - `[chains.optimism.rpc-url]` - the RPC endpoint to use for mainnet Optimism (e.g. `https://mainnet.optimism.io`)
+    - `[account.path]` - the path to your JSON keystore file
+    - `[account.password]` - the password to unlock your keystore file
+    - `[chains.l1.rpc-url]` - the RPC endpoint to use for mainnet Ethereum (e.g. `https://mainnet.infura.io/v3/ID`)
```
Comment (fredo) [inline README.md:77]: the key for l1 is now called `base-chain`

File: README.md [hunk 5]
```diff
@@ -60,21 +63,20 @@ If it is not, you can apply for whitelisting at hello@beamerbridge.com.
...
+    - `[chains.arbitrum.rpc-url]` - the RPC endpoint to use for mainnet Arbitrum (e.g. `https://arb1.arbitrum.io/rpc`)
+    - `[chains.optimism.rpc-url]` - the RPC endpoint to use for mainnet Optimism (e.g. `https://mainnet.optimism.io`)
```
Comment (fredo) [inline README.md:79]: would be nice to also add chains.ethereum...

File: README.md [hunk 6]
```diff
@@ -85,17 +87,16 @@ If it is not, you can apply for whitelisting at hello@beamerbridge.com.
...
    sure that the following keys have correct values:
 
-   - `[account.path]` - the path to your JSON keystore file
-   - `[account.password]` - the password to unlock your keystore file
-   - `[chains.l1.rpc-url]` - the RPC endpoint to use for Goerli Ethereum
-   - `[chains.arbitrum.rpc-url]` - the RPC endpoint to use for Goerli Arbitrum
-   - `[chains.optimism.rpc-url]` - the RPC endpoint to use for Goerli Optimism
+    - `[account.path]` - the path to your JSON keystore file
+    - `[account.password]` - the password to unlock your keystore file
+    - `[chains.l1.rpc-url]` - the RPC endpoint to use for Goerli Ethereum
```
Comment (fredo) [inline README.md:97]: same as the above

File: README.md [hunk 7]
```diff
@@ -110,13 +111,13 @@ If it is not, you can apply for whitelisting at hello@beamerbridge.com.
 1. For mainnet, run:
 
    ```
-   docker compose -f docker-compose-mainnet.yml up -d
+   docker compose -f docker-compose-mainnet.yml up -d mainnet-agent
```
Comment (fredo) [inline README.md:114]: why do we add `mainnet-agent` here?

File: README.md [hunk 8]
```diff
@@ -140,5 +141,13 @@ You are now running an agent for the Beamer bridge. Thank you and please
 For more information on running an agent and updating to new agent versions,
 please refer to [documentation](https://docs.beamerbridge.com/running.html).
 
-We recommend that you provide your own monitoring. Setting up agent monitoring
-is out of scope of this document.
+## Beamer health-check
+
+The beamer executable includes a health-check command that can be used to
+analyze the beamer contracts and also reports agent liquidity. The health-check command
+fetches all the event executed by the contracts and analyzes them to determine
```
[the remark this answers is not in the document]
Comment (fredo) [reply README.md:148]:
```suggestion
fetches all the events emitted by the contracts and analyzes them to determine
```

**Comment by bilbeyt** [comment]
We should absolutely mention about poll-period and that should be float, also confirmation-blocks as int

File: README.md (hunk 4 above)
Comment (manuelwedler) [reply README.md:77]: Then I think the `agent.conf.example` in the main beamer repo is also outdated.

File: docker-compose-mainnet.yml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 version: '3.7'
 
 x-defaults: &defaults
-  image: ghcr.io/beamer-bridge/beamer-agent:v1.0.3
+  image: ghcr.io/beamer-bridge/beamer:v1.0.2-31-gee4fef6
```
Comment (fredo) [inline docker-compose-mainnet.yml]: shouldn't this be v2.0.0

Comment (manuelwedler) [reply docker-compose-mainnet.yml]: Think you are not checking the newest commit

File: docker-compose-mainnet.yml [hunk 10]
```diff
@@ -15,4 +15,24 @@ services:
     << : *defaults
     restart: always
     # the agent command is included in the ENTRYPOINT declaration
-    command: --config /data/agent-mainnet.conf --source-chain optimism --target-chain arbitrum
+    command: agent --config /data/agent-mainnet.conf --source-chain optimism --target-chain arbitrum
```
[the remark this answers is not in the document]
Comment (fredo) [reply docker-compose-mainnet.yml]: the respective source and target chain event processors are created out of the config

[the remark this answers is not in the document]
Comment (manuelwedler) [reply docker-compose-mainnet.yml]: I removed them in the latest commit

File: README.md (hunk 7 above)
Comment (manuelwedler) [reply README.md:114]: @compojoom ?

File: README.md [hunk 11]
```diff
@@ -145,7 +147,7 @@ please refer to [documentation](https://docs.beamerbridge.com/running.html).
 
 The beamer executable includes a health-check command that can be used to
 analyze the beamer contracts and also reports agent liquidity. The health-check command
-fetches all the event executed by the contracts and analyzes them to determine
+fetches all the events executed by the contracts and analyzes them to determine
```
Comment (fredo) [inline README.md:150]: nitpick: emitted is fitting better instead of executed

File: README.md (hunk 7 above)
Comment (bilbeyt) [reply README.md:116]: i can answer as service name is mainnet-agent in the docker-compose if we use this only agent runs and not health-check

File: data/agent-goerli.conf [hunk 12]
```diff
@@ -2,17 +2,24 @@ log-level = "debug"
...
+[chains.base-chain]
 rpc-url = ""
 
+[chains.goerli]
+rpc-url = ""
+confirmation-blocks = 1
+
 [chains.optimism]
 rpc-url = ""
+poll-period = 5.0
```
Comment (bilbeyt) [inline data/agent-goerli.conf:22]: nitpick: doesnt change anything already defined as default above so we can remove this line.

File: data/agent-mainnet.conf [hunk 13]
```diff
@@ -2,30 +2,46 @@ log-level = "debug"
...
+[chains.ethereum]
+rpc-url = ""
+confirmation-blocks = 1
+
 [chains.arbitrum]
 rpc-url = ""
 
 [chains.optimism]
 rpc-url = ""
+poll-period = 5.0
```
Comment (bilbeyt) [inline data/agent-mainnet.conf:25]: same as above

**Review by fredo — APPROVED** [review]
nice

File: data/agent-goerli.conf (hunk 12 above)
Comment (manuelwedler) [reply data/agent-goerli.conf:22]: Fred had it in his PR, I think as an example

File: data/agent-mainnet.conf [hunk 14]
```diff
@@ -2,30 +2,46 @@ log-level = "debug"
...
+confirmation-blocks = 0
 
 [account]
 # make sure that the filename matches your keystore file
 path = "/data/account/keystore.json"
 password = ""
 
-[chains.l1]
+[chains.base-chain]
```
Comment (bilbeyt) [inline data/agent-mainnet.conf:13]: just base-chain. btw not chains.base-chain

File: data/agent-goerli.conf [hunk 15]
```diff
@@ -2,17 +2,24 @@ log-level = "debug"
...
+confirmation-blocks = 0
 
 [account]
 # make sure that the filename matches your keystore file
 path = "/data/account/keystore.json"
 password = ""
 
-[chains.l1]
+[chains.base-chain]
```
Comment (bilbeyt) [inline data/agent-goerli.conf:13]: this also just base-chain

**Review by fredo — APPROVED** [review]

**Review by bilbeyt — APPROVED** [review]
nice

**Pull request merged**